### PR TITLE
feat(discriminator.querying): allows update doc with discriminatorKey

### PR DESCRIPTION
**Pull request by abarriel**
https://github.com/Automattic/mongoose/issues/13055

### CODE REVIEW

File: test/model.discriminator.querying.test.js [hunk 1]
```diff
@@ -30,15 +30,15 @@ const ImpressionEventSchema = new BaseSchema({ element: String });
 const ConversionEventSchema = new BaseSchema({ revenue: Number });
 const SecretEventSchema = new BaseSchema({ secret: { type: String, select: false } });
 
-describe('model', function() {
-  describe('discriminator()', function() {
+describe('model', function () {
```
Comment (vkarpov15) [inline test/model.discriminator.querying.test.js:33]: Can you please undo these formatting changes? They're likely to cause merge conflicts for us, and they make this PR more difficult to review.

**Review by vkarpov15 — CHANGES_REQUESTED** [review]
LGTM overall. Just please undo the formatting changes.

File: test/model.discriminator.querying.test.js (hunk 1 above)
Comment (abarriel) [reply test/model.discriminator.querying.test.js]: My bad, fixed

**Review by vkarpov15 — APPROVED** [review]
Thanks :+1:

**Pull request merged**